packages/instrumentor/instrument.ts

```diff
@@ -14,16 +14,40 @@
  * limitations under the License.
  */
 
-import { transformSync } from "@babel/core";
-import { hookRequire } from "istanbul-lib-hook";
+import { PluginItem, transformSync } from "@babel/core";
+import { hookRequire, TransformerOptions } from "istanbul-lib-hook";
 import { codeCoverage } from "./plugins/codeCoverage";
 import { compareHooks } from "./plugins/compareHooks";
+import { functionHooks } from "./plugins/functionHooks";
+import { hookManager } from "@jazzer.js/hooking";
 
 export function registerInstrumentor(includes: string[], excludes: string[]) {
-	hookRequire(shouldInstrument(includes, excludes), instrumentCode);
+	const shouldInstrument = shouldInstrumentFn(includes, excludes);
+	hookRequire(
+		() => true,
```

Comment:
Could we extract the checks from below into a predicate that can determine if a file should be transformed or not (coverage and hooks)? Then we could use a fixed list of plugins.

Reply:
I thought about something like: 
```
hookRequire(
  (filename) => shouldInstrument(filename) || hookManager.hasFunctionsToHook(filename),
``` 
And adding a fixed set of plugins. Every plugin would then need to check if the given file should be transformed.
This would also potentially save internal work in `istanbul-lib-hook`  and wouldn't even call the transformer. But I took a look at the library code and there is not much that could be skipped. So, it's probably ok as it is now.